### ISSUE TITLE
Notification component's role="alert" creates accessibility issue for screen reader users

### ISSUE THREAD

**Issue by mellis481** [ant-design/ant-design#39568]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/e9evzr)

### Steps to reproduce

1. Open example.
2. Click button to open notification box.

### What is actually happening?

Since `role="alert"` is used, the notification interrupt whatever the user is doing. This also stops a screen reader from finishing whatever it was saying before the alert interrupted it (and will not pick up where it left off) which can make a user feel stranded. Toast-style notifications should be used for non-critical, non-interactive content so it's inappropriate to use `role="alert"` which implicitly has an `aria-live` value of `assertive`.

### What is expected?

Instead, `role="status"` is more appropriate as it allows for current screen reader announcements to be read before announcing the non-critical notification content. This requires the notification elements to be with an element with `role="region"` and `aria-live="polite"`.

| Environment | Info |
| --- | --- |
| antd | 4.24.5 |
| React | React |
| System | Windows 10 |
| Browser | All |

---
More information:
- https://www.scottohara.me/blog/2019/07/08/a-toast-to-a11y-toasts.html#inclusive-ux-of-a-toast
- https://adrianroselli.com/2020/01/defining-toast-messages.html#NeededFeatures

**Comment by mellis481** [comment]
Would a PR for this be accepted?

**Comment by MadCcc** [comment]
Cannot find anything different, could you provide more detail for `role="alert"`?

**Comment by dgreene1** [comment]
@MadCcc I don’t think it’s @mellis481’s responsibility to show the problem when the links he provided describes it perfectly.

So, if we can’t agree that role=alert is problematic, then I think we can agree that both should be options. That way the default behavior can persist but consumers like us can choose the more polite behavior.

Is that a more acceptable solution?

**Comment by mellis481** [comment]
> Cannot find anything different, could you provide more detail for `role="alert"`?

@MadCcc The difference is succinctly described in the "What is actually happening?" and "What is expected?" sections above.

**Comment by Thulof** [comment]
Hi, @mellis481 @dgreene1 @MadCcc I have pushed a PR to add a `role` prop to notification component, please review it. https://github.com/ant-design/ant-design/pull/39611

**Comment by mellis481** [comment]
@Thulof Can you check my latest comment in your PR please?

**Comment by Thulof** [comment]
> @Thulof Can you check my latest comment in your PR please?

Sure, I will check it.

In the past period, I had a bad cold, so I didn't follow on this issue. I am sorry.

**Comment by mellis481** [comment]
@Thulof Any update?

**Comment by guan404ming** [comment]
I think I could help to close this issue

**Comment by mellis481** [comment]
@guan404ming If you're talking about creating a PR for it, that would be awesome. Thanks!

**Comment by guan404ming** [comment]
This issue should be closed~~ @li-jia-nan
